### PR TITLE
Bill discounting revamp: Bill Discounting duplicate Invites Fix

### PR DESCRIPTION
**Pull request by ShreyDhyani**

### CODE REVIEW

File: app/controllers/billDiscountSuppliers.js [hunk 1]
```diff
@@ -88,16 +93,24 @@ module.exports = {
...
+          profile.uuid,
+          item.profile_uuid
+        );
+        if (inviteStatus === "pending") {
+          pendingCount = pendingCount + 1;
+        }
+        if (!inviteExists) {
+          inviteCount = inviteCount + 1;
+          supplier = await upsertInvite(profile.uuid, item.profile_uuid, item);
+        } else {
```
Comment (rajeshn95) [inline app/controllers/billDiscountSuppliers.js:110]: @ShreyDhyani Is this else block required here?

File: app/controllers/billDiscountSuppliers.js [hunk 2]
```diff
@@ -88,16 +93,24 @@ module.exports = {
...
 
-        let supplier = await BillDiscountSuppliers.create({
-          invited_by: profile.uuid,
-          status: "pending",
-          ...item,
-        });
+        let supplier = null;
+        const [inviteExists, inviteStatus] = await checkExistingInvites(
+          profile.uuid,
+          item.profile_uuid
```
Comment (rajeshn95) [inline app/controllers/billDiscountSuppliers.js:102]: @ShreyDhyani What happens when email/mobile is not onboarded in wiredup?

File: app/functions/neptune/neptuneCaller.js [hunk 3]
```diff
@@ -3,7 +3,7 @@ var uuid = require("uuid");
 var dayjs = require("dayjs");
 
 module.exports = async function (payload) {
-  console.log("checkherepayloadinneptunecaller", JSON.stringify(payload));
+  // console.log("checkherepayloadinneptunecaller", JSON.stringify(payload));
```
Comment (rajeshn95) [inline app/functions/neptune/neptuneCaller.js:6]: @ShreyDhyani remove this comment

**Review by rajeshn95 — CHANGES_REQUESTED** [review]

File: app/controllers/billDiscountSuppliers.js (hunk 2 above)
Comment (ShreyDhyani) [reply app/controllers/billDiscountSuppliers.js:102]: So here we need to handle the case if the invited user is not on wired up?? but we still need to send an invite???

File: app/controllers/billDiscountSuppliers.js (hunk 1 above)
Comment (ShreyDhyani) [reply app/controllers/billDiscountSuppliers.js:110]: Removed

File: app/controllers/billDiscountSuppliers.js (hunk 2 above)
Comment (rajeshn95) [reply app/controllers/billDiscountSuppliers.js]: @ShreyDhyani Yes

**Review by rajeshn95 — APPROVED** [review]

**Pull request merged**